### DISCUSSION TITLE
Custom Typing Practice Lists

### DISCUSSION THREAD

**Discussion by DhananjayPorwal** [monkeytypegame/monkeytype#5720] [Ideas] [answered]
**Description:**
Implement a feature that allows users to upload a text file containing a custom list of words or paragraphs for typing practice. This feature will enable users to practice typing with content that is relevant to their interests or learning goals.

**Details:**
- **File Upload:** Users can upload a text file with a minimum of 200 words.
- **Content Restrictions:** The uploaded file must contain at least one paragraph.
- **Learning Integration:** For example, a user interested in Machine Learning can upload a text file with definitions or articles related to Machine Learning. This will help users subconsciously learn about their topics of interest while practicing typing.

**Impact:**
- **Personalized Learning:** Users can integrate their typing practice with learning new concepts or reinforcing knowledge in specific areas.
- **Enhanced Engagement:** Practicing with relevant content can increase user engagement and motivation.
- **Skill Improvement:** This feature can help users improve both their typing speed and their understanding of specific subjects simultaneously.

**Comment by fehmer** [comment, marked as answer]
- pick test mode `custom`
- hit change
- click `open file`
- set mode to `random`
- click `ok`

**Reply by DhananjayPorwal** [reply]
Thanks, I didn’t know this function existed before.
